### ISSUE TITLE
Layout/EmptyLinesAroundAccessModifier false positive on call to method named public

### ISSUE THREAD

**Issue by gaffneyc** [rubocop-hq/rubocop#7604]
We have an ActiveRecord model which has a boolean attribute named public. When doing a recent upgrade from Rubocop 0.58 to 0.78 we are now receiving a warning that we need to add empty lines around the `public` access modifier even though it is a method call.

--------

## Expected behavior

Attributes or methods that are named `public`, `private`, or `protected` should not trigger the EmptyLinesArroundAccessModifier Cop

## Actual behavior

```
Layout/EmptyLinesAroundAccessModifier: Keep a blank line before and after public.
      public { false }
      ^^^^^^
```

## Steps to reproduce the problem

```ruby
FactoryBot.define do
  factory :thing do
    trait :private do
      public { false }
    end
  end
end
```

## RuboCop version

```
$ bundle exec rubocop -V
0.78.0 (using Parser 2.7.0.0, running on ruby 2.6.5 x86_64-linux)
```

**Comment by Tietew** [comment]
Rubocop cannot detect wether the `public` is a part of FactoryBot's factory or not.
It should be excluded explicitly in project's .rubocop.yml.

For example:
``` yaml
Layout/EmptyLinesAroundAccessModifier:
  Exclude:
    - spec/factories/*
```

**Comment by gaffneyc** [comment]
Rubocop may not be able to determine the context but I assume it should be able to determine if the usage of public (or private or protected) matches usage as an access modifier. In this case, `Module.public` doesn't take a block so it could be excluded from the cop.

As far as I understand there are only a couple forms of `public`, `private`, `protected` that are valid. `public` with no arguments, `public(*symbols)`, and `public(*strings)`.

```ruby
class Example
  public def foo
    "bar"
  end

  public

  def bar
  end

  def baz
  end
  private :baz

  def biz
  end
  protected "biz"
end
```

**Comment by gaffneyc** [comment]
Tested with 0.88.0 and it's still an issue.

**Comment by gaffneyc** [comment]
Appears to no longer be an issue as of 0.92.0
